refactron/rag/parser.py

```
@@ -0,0 +1,259 @@
+    module_docstring: Optional[str]
+
+
+class CodeParser:
+    """AST-aware code parser using tree-sitter."""
+
+    def __init__(self):
+        """Initialize the parser."""
+        if not TREE_SITTER_AVAILABLE:
+            raise RuntimeError(
+                "tree-sitter is not available. Install with: pip install tree-sitter tree-sitter-python"
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)